bats/core/api/auth.bats

```diff
@@ -0,0 +1,237 @@
+  flowId=$(curl_output '.result')
+  [[ "$flowId" =~ ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$ ]] || exit 1
+
+  exec_graphql 'charlie' 'identity'
+  [[ "$(graphql_output '.data.me.email.address')" == "null" ]] || exit 1
+  [[ "$(graphql_output '.data.me.totpEnabled')" == "false" ]] || exit 1
+
+  count=$(getEmailCount "$email")
+  [[ "$count" -eq "$countInit" ]] || exit 1
+
+  # TODO: email to the sender highlighting the email was removed
+}
+
+@test "auth: remove phone login" {
+  email=$(read_value 'charlie.email')
+
+  variables="{\"input\": {\"email\": \"$email\"}}"
+  exec_graphql 'charlie' 'user-email-registration-initiate' "$variables"
+  emailRegistrationId="$(graphql_output '.data.userEmailRegistrationInitiate.emailRegistrationId')"
+
+  code=$(getEmailCode "$email")
+  echo "The code is: $code"
+
+  variables="{\"input\": {\"code\": \"$code\", \"emailRegistrationId\": \"$emailRegistrationId\"}}"
+  exec_graphql 'charlie' 'user-email-registration-validate' "$variables"
+  [[ "$(graphql_output '.data.userEmailRegistrationValidate.me.email.address')" == "$email" ]] || exit 1
+  [[ "$(graphql_output '.data.userEmailRegistrationValidate.me.email.verified')" == "true" ]] || exit 1
+
+  # Remove phone.
+  exec_graphql "charlie" "user-phone-delete"
+  [[ "$(graphql_output '.data.userPhoneDelete.me.phone')" == "null" ]] || exit 1
+}
+
+@test "auth: adding totp" {
+  authToken=$(read_value 'charlie')
+
+  # Initiate TOTP Registration
+  exec_graphql 'charlie' 'user-totp-registration-initiate'
+
+  totpRegistrationId="$(graphql_output '.data.userTotpRegistrationInitiate.totpRegistrationId')"
+  [[ "$totpRegistrationId" =~ ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$ ]] || exit 1
+
+  totpSecret="$(graphql_output '.data.userTotpRegistrationInitiate.totpSecret')"
+  [ -n "$totpSecret" ] || exit 1
+
+  # Validate TOTP Registration
+  totpCode=$(generateTotpCode "$totpSecret")
+  variables="{\"input\": {\"totpCode\": \"$totpCode\", \"totpRegistrationId\": \"$totpRegistrationId\", \"authToken\": \"$authToken\"}}"
+  exec_graphql 'charlie' 'user-totp-registration-validate' "$variables"
+
+  # Checking the response structure
+  totpEnabled="$(graphql_output '.data.userTotpRegistrationValidate.me.totpEnabled')"
+  [ "$totpEnabled" == "true" ] || exit 1
+}
+
+@test "auth: log in with email with totp activated" {
+  skip "This test needs to be fixed"
+
+  email=$(read_value 'charlie.email')
+
+  # code request
+  variables="{\"email\": \"$email\"}"
+  curl_request "http://${GALOY_ENDPOINT}/auth/email/code" "$variables"
+  emailLoginId="$(curl_output '.result')"
+  [ "$emailLoginId" != "null" ]
+
+  code=$(getEmailCode "$email")
+  [ "$code" != "" ]
+
+  # validating email with code
+  variables="{\"code\": \"$code\", \"emailLoginId\": \"$emailLoginId\"}"
+  curl_request "http://${GALOY_ENDPOINT}/auth/email/login" "$variables"
+  authToken="$(curl_output '.result.authToken')"
+  [ -n "$authToken" ] || exit 1
```

Comment:
I can't seem to figure out why this one fails here with the `curl_output` being `{"errors": ""}`